src/ore/src/tracing.rs

```diff
@@ -315,16 +316,51 @@ where
             .with_exporter(exporter)
             .install_batch(opentelemetry::runtime::Tokio)
             .unwrap();
+
+        // By default we turn off tracing from the following crates, because they
+        // have long-lived Spans, which OpenTelemetry does not handle well.
+        //
+        // Note: folks should feel free to add more crates here if we find more
+        // with long lived Spans.
+        let default_targets = [("h2", LevelFilter::OFF), ("hyper", LevelFilter::OFF)];
+
         let (filter, filter_handle) = reload::Layer::new(if otel_config.start_enabled {
-            otel_config.filter
+            let new_targets = Targets::default()
+                .with_targets(default_targets)
+                .with_targets(&otel_config.filter);
+
+            if let Some(default_level) = otel_config.filter.default_level() {
+                new_targets.with_default(default_level)
+            } else {
+                new_targets
+            }
         } else {
             // The default `Targets` has everything disabled.
             Targets::default()
         });
         let layer = tracing_opentelemetry::layer()
+            // OpenTelemetry does not handle long-lived Spans well, and they end up continuously
+            // eating memory until OOM. So we set a max number of events that are allowed to be
+            // logged to a Span, once this max is passed, old events will get dropped
+            //
+            // TODO(parker-timmerman|guswynn): make this configurable with LaunchDarkly
+            .max_events_per_span(2048)
```

Comment:
note: after chatting with Nikhil offline we opted to bump this up from 256 to something higher to handle the case where we enabled tracing with the `TRACE` level. Testing this setup locally we seem to be using ~200Kb for events, which IMO is quite reasonable.